app/models/spree/order.rb

```diff
@@ -756,11 +756,19 @@ def associate_customer
     def ensure_customer
       return if associate_customer
 
-      customer_name = bill_address.andand.full_name
-      self.customer = Customer.create(enterprise: distributor, email: email_for_customer,
-                                      user: user, name: customer_name,
-                                      bill_address: bill_address.andand.clone,
-                                      ship_address: ship_address.andand.clone)
+      self.customer = Customer.new(
+        enterprise: distributor,
+        email: email_for_customer,
+        user: user,
+        name: bill_address.andand.full_name,
+        bill_address: bill_address.andand.clone,
+        ship_address: ship_address.andand.clone
+      )
+      customer.save
+
+      unless customer.valid?
```

Comment:
Very minor point but I'd write this as a one-liner with `if customer.errors.any?` since we expect the customer to be valid.

Reply:
I pushed-forced a new commit making this conditional "positive" without having to couple to `errors`